plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/commands/NudgeOnRemovalCommand.java

```diff
@@ -228,7 +238,7 @@ private int getDefaultLifelineTop(MLifeline lifeline) {
 				/*
 				 * keep previous distance between delete range and next element and move below deleted range
 				 */
-				delta = firstBottomAboveDeletion.orElse(bottomFinal) - bottomFinal
+				delta = firstBottomAboveDeletion.orElse(deletedTop) - bottomFinal
```

Comment:
As far as I can see the bug was, that we have no benchmark (first element on lifeline), then got an empty firstBottomAboveDeletion, and the orElse was bottomFinal, resulting in no move. 

Would it work to not put any additional elements in the bottomToElements-map but computing the orElse value lazily?

Reply:
> As far as I can see the bug was, that we have no benchmark (first element on lifeline), then got an empty firstBottomAboveDeletion, and the orElse was bottomFinal, resulting in no move.

Exactly!

> Would it work to not put any additional elements in the bottomToElements-map but computing the orElse value lazily?
Yes, that should work too. I had this implementation first but felt that putting the lowest related lifeline into the map would be the more general solution, as the bottom of the lowest related lifeline should act as a benchmark just as any other element above would -- so except for adding it, there would be no special case handling involved. But if you disagree, we can handle it only as a special case in the orElse(...).

Reply:
I was asking because I am not 100% sure that this will not confuse cases with nudged lifelines and more complex deletions involving multiple elements. But this is just speculation, so I guess it is enough to keep this in mind just in case.